src/main/java/org/mastodon/mamut/feature/branch/sinuosity/BranchSinuosityFeatureComputer.java

```diff
@@ -0,0 +1,105 @@
+import org.scijava.ItemIO;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+/**
+ * Computes {@link BranchSinuosityFeature}
+ */
+@Plugin( type = MamutFeatureComputer.class )
+public class BranchSinuosityFeatureComputer implements MamutFeatureComputer
+{
+
+	@Parameter
+	private ModelBranchGraph branchGraph;
+
+	@Parameter( type = ItemIO.OUTPUT )
+	private BranchSinuosityFeature output;
+
+	@Override
+	public void createOutput()
+	{
+		if ( null == output )
+			output = new BranchSinuosityFeature( new DoublePropertyMap<>( branchGraph.vertices().getRefPool(), 0 ) );
+	}
+
+	@Override
+	public void run()
+	{
+		computeBranchSinuosity();
+	}
+
+	private void computeBranchSinuosity()
+	{
+		DepthFirstIterator< BranchSpot, BranchLink > depthFirstIterator = new DepthFirstIterator<>( branchGraph );
+		for ( BranchSpot root : RootFinder.getRoots( branchGraph ) )
+		{
+			int numDimensions = root.numDimensions();
+			final double[] currentSpotCoordinates = new double[ numDimensions ];
+			double[] previousSpotCoordinates = new double[ numDimensions ];
+
+			depthFirstIterator.reset( root );
+			boolean isRoot = true;
+			while ( depthFirstIterator.hasNext() )
+			{
+				BranchSpot currentBranchSpot = depthFirstIterator.next();
+				double[] branchStartCoordinates = new double[ 3 ];
+				boolean isBranchStartCoordinatesSet = false;
+				boolean isPreviousCoordinatesSet = false;
+				Spot currentSpot;
+				double totalDistanceDuringCellLifeCycle = 0d;
+				final Iterator< Spot > it = branchGraph.vertexBranchIterator( currentBranchSpot );
+				while ( it.hasNext() )
+				{
+					currentSpot = it.next();
+					currentSpot.localize( currentSpotCoordinates );
+					if ( !isBranchStartCoordinatesSet )
+					{
+						if ( isRoot )
+						{
+							branchStartCoordinates =
+									Arrays.copyOf( currentSpotCoordinates, currentSpotCoordinates.length );
+						}
+						else
+						{
+							for ( int d = 0; d < numDimensions; d++ )
+							{
+								branchStartCoordinates[ d ] =
+										currentBranchSpot.incomingEdges().iterator().next().getSource()
+												.getDoublePosition( d );
+							}
+						}
+						isBranchStartCoordinatesSet = true;
+					}
+					if ( isPreviousCoordinatesSet )
+						totalDistanceDuringCellLifeCycle +=
+								GeometryUtils.getEuclideanDistance( currentSpotCoordinates, previousSpotCoordinates );
+					else
+						totalDistanceDuringCellLifeCycle +=
+								GeometryUtils.getEuclideanDistance( currentSpotCoordinates, branchStartCoordinates );
+					previousSpotCoordinates = Arrays.copyOf( currentSpotCoordinates, currentSpotCoordinates.length );
+					isPreviousCoordinatesSet = true;
+				}
+				double directDistanceDuringCellLifeCycle =
+						GeometryUtils.getEuclideanDistance( branchStartCoordinates, currentSpotCoordinates );
+				double sinuosity = totalDistanceDuringCellLifeCycle / directDistanceDuringCellLifeCycle;
```

Comment:
Extracted to a separate and simplified method.
Partially, the complication of the previous implementation occurred, since I wanted to compute the sinuosity including the state before the cell division.

Reply:
Now that method is simpler that's good.